src/web/mjs/connectors/CoroCoro.mjs

```diff
@@ -0,0 +1,105 @@
+import Chapter from '../engine/Chapter.mjs';
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+const weekdays = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'];
+
+function unhex(hex) {
+    return new Uint8Array(hex.match(/../g).map(byte => parseInt(byte, 16)));
+}
+
+export default class CoroCoro extends Connector {
+    constructor() {
+        super();
+        super.id = 'corocoro';
+        super.label = '週刊コロコロコミック (Shuukan CoroCoro Comic)';
+        this.tags = ['manga', 'japanese'];
+        this.url = 'https://www.corocoro.jp';
+    }
+
+    /**
+     * @param {URL} uri
+     */
+    async _getMangaFromURI(uri) {
+        if (uri.pathname.startsWith('/title/')) {
+            const id = uri.pathname.substring('/title/'.length);
+            /** @type {HTMLHeadingElement[]} */
+            const data = await this.fetchDOM(uri, 'h1');
+            const title = data[0].innerText.trim();
+            return new Manga(this, id, title);
+        }
+        if (uri.pathname.startsWith('/chapter/')) {
+            const scripts = await this.fetchDOM(uri, 'script');
+            const script = scripts[scripts.length - 1].innerText;
+            /** @type {string} */
+            const json = eval(script.substring(22, script.length - 2));
+            /** @type {{chapterListSection: {titleName: string}, viewerSection: {titleID: number}}} */
+            const data = JSON.parse(json.substring(json.indexOf(':') + 1))[3]['children'][0][3];
+            const id = data.viewerSection.titleID.toString();
+            const title = data.chapterListSection.titleName;
+            return new Manga(this, id, title);
+        }
+    }
+
+    /** @type {Connector['_getMangas']} */
+    async _getMangas() {
+        /** @type {HTMLScriptElement[]} */
+        const scripts = await this.fetchDOM(this.url + '/rensai', 'script');
+        const script = scripts[scripts.length - 2].innerText;
+        /** @type {string} */
+        const json = eval(script.substring(22, script.length - 2));
```

Comment:
not fond of evaluating code from website directly inside our code, for security reasons.